core/client/src/main/java/alluxio/client/file/FileSystemUtils.java

```diff
@@ -161,7 +163,12 @@ public static long persistFile(FileSystem fs, AlluxioURI uri, URIStatus status,
       if (!ufs.exists(parentPath) && !ufs.mkdirs(parentPath, true)) {
         throw new IOException("Failed to create " + parentPath);
       }
-      OutputStream out = closer.register(ufs.create(dstPath.getPath()));
+      // TODO(chaomin): should also propagate ancestor dirs permission to UFS.
+      URIStatus uriStatus = fs.getStatus(uri);
+      Permission perm = new Permission(uriStatus.getUserName(), uriStatus.getGroupName(),
+          (short) uriStatus.getPermission());
```

Comment:
Yes we should. FYI another PR https://github.com/Alluxio/alluxio/pull/3566 did the rename from permission->mode. I'll update once that is merged.